src/main/scala/swiss/dasch/domain/AssetModel.scala

```diff
@@ -51,20 +51,27 @@ final case class SimpleAsset(id: AssetId, belongsToProject: ProjectShortcode) ex
   def makeImageAsset(
     originalFilename: NonEmptyString,
     original: OriginalFile,
-    derivative: JpxDerivativeFile
-  ): ImageAsset =
-    ImageAsset(id, belongsToProject, originalFilename, original, derivative)
+    derivative: DerivativeFile
+  ): ComplexAsset.ImageAsset =
+    ComplexAsset.ImageAsset(id, belongsToProject, originalFilename, original, derivative)
 }
 
-final case class ImageAsset(
-  id: AssetId,
-  belongsToProject: ProjectShortcode,
-  originalFilename: NonEmptyString,
-  original: OriginalFile,
-  derivative: JpxDerivativeFile
-) extends Asset {
-  def originalInternalFilename: String = original.filename
-  def derivativeFilename: String       = derivative.filename
+sealed trait ComplexAsset extends Asset {
+  def originalFilename: NonEmptyString
+  def original: OriginalFile
+  def derivative: DerivativeFile
+  final def originalInternalFilename: String = original.filename
+  final def derivativeFilename: String       = derivative.filename
+}
+object ComplexAsset {
+  final case class ImageAsset(
+    id: AssetId,
+    belongsToProject: ProjectShortcode,
+    originalFilename: NonEmptyString,
+    original: OriginalFile,
+    derivative: DerivativeFile
```

Comment:
The type of the `derivative` property of an `ImageAsset` must be `JpxDerivativeFile`. This change this would allow an `ImageAsset` to have a `DerivativeFile` which is a `*.xls` file for example.

In general I am not yet fully on board with the idea of creating a `ComplexAsset` type from which all other assets should derive from, we already have an `Asset` trait binding all assets together. What is `ComplexAsset` representing as opposed to an `Asset` why shouldn't we expand the `Asset` trait instead? 

I suppose you are trying to model the fact that certain metadata for assets is available for all assets. All the meta information of original and derivative files could also become its own property. The `SimpleAsset` maybe is rather an `AssetRef` an may not necessarily extend `Asset`, it represents the smallest amount of information for a pointer to any _potentially existing_ asset in the filesystem and currently is only used to create an `Asset`. More specific assets should be their own case class, like the `ImageAsset`. IMHO instances of these should reference an _existing_ asset in the filesystem.

I also do not yet fully understand if all assets always have or should have an original *and* a derivative, this is pretty tight coupling.  For example do pdfs, xls files have a derivative and if how is it different from the original?

Reply:
My message in chat might have made some of that a bit clearer (sorry for the multi-channel communication!) - mostly that I am aware that this part of the code is not yet ideal and that I drafted it this way as a canvas for further improvement (the previous implementation couples Asset too directly to image implementation details, like the Jpx stuff, so something needed to be changed).  
Adding `ComplexAsset` was really just in differentiation to `SimpleAsset` but I would definitely prefer only to have `Asset` and `AssetRef` like you propose. (And maybe we should consider re-naming `Asset` to something like `AssetInfo` or `AssetMetadata`, as it does not actually hold the asset itself.)  
And I'm all in favour of not over-complicating things.

Regarding assets always having  both original and derivative: As long as Ingest implements the behaviour that we currently have with SIPI, we're bound to how SIPI does things. And SIPI has an original and a derivative for all asset types, even though for most types, those two files are identical copies of each other - which can be verified by the fact that they have identical checksums in the .info files. (I was a bit irritated by that too, but we checked and that's how SIPI does it. if you upload `foo.txt`, sipi will create `<asset-id>.txt` and `<asset-id>.txt.orig` and store information on both of them in the .info file.)

As I'm not working today and tomorrow, the following are our options:
- this PR waits until Monday, but probably becomes obsolete (which I don't mind) by the work that @mpro7 and you do in the meanwhile, as ingest is our current highest working priority
- we merge this as it is and it can be improved in a next PR (it's not currently in a broken state)
- one of you two - or you together - take this PR over and improve it until it's good to merge

I'm happy with either option and since I'm off today, I would also leave the choice to you.

Reply:
This PR is also a consequence of our work / brainstorming together yesterday on how to start to implement mass upload into ingest service. As we decided, we will go with baby steps, and this PR is one of those.

Naming could be always improved, one could always prefer something else over proposed stuff, but as mentioned above, it's not breaking the service and shouldn't stop also the further work.